File: loop_transformations/sources/apply_span.1.c
/*
* @@name:	apply_span.1
* @@type:	C
* @@operation:	compile
* @@expect:	success
* @@version:	omp_6.0
*/
void span_apply(double A[128][128])
{
   #pragma omp for collapse(2)
   #pragma omp tile sizes(16,16) \
               apply(grid: interchange,reverse)
   for (int i = 0; i < 128; ++i)
      for (int j = 0; j < 128; ++j)
         A[i][j] = A[i][j] + 1;
}

File: loop_transformations/sources/apply_span.1.f90
! @@name:       apply_span.1
! @@type:       F-free
! @@operation:  compile
! @@expect:     success
! @@version:    omp_6.0
subroutine span_apply( A )
   implicit none
   double precision :: A(0:127,0:127)
   integer          ::   i   , j

    !$omp for collapse(2)
    !$omp tile sizes(16,16) apply(grid: interchange,reverse)
    do i = 0, 127
    do j = 0, 127
      A(j,i) = A(j,i) + 1
    enddo; enddo

end subroutine
